src/linear_regressor.py

```diff
@@ -0,0 +1,79 @@
+import math
+from src.measurement import Measurement
+
+class LinearRegression:
+    """
+    Stores a linear regression of the form y = m * x + n
+    """
+
+    def __init__(self, m: float, m_error: float, n: float, n_error: float) -> None:
+        self.m = m
+        self.m_error = m_error
+        self.n = n
+        self.n_error = n_error
+
+    def best_fit(self):
+        return LinearFunction(self.m, self.n)
+    
+    def error_curve_high_slope(self):
+        return LinearFunction(self.m + self.m_error, self.n - self.n_error)
+    
+    def error_curve_low_slope(self):
+        return LinearFunction(self.m - self.m_error, self.n + self.n_error)
+    
+class LinearFunction:
+    """
+    Stores a linear function of the form y = m * x + n
+    """
+
+    def __init__(self, m: float, n: float) -> None:
+        self.m = m
+        self.n = n
+
+    def eval(self, x: float) -> float:
+        return self.m * x + self.n
+
+def do_linear_regression(measurements: list[Measurement]):
+    """
+    Makes a linear regression.
+    """
+    x = [m.x for m in measurements]
+    y = [m.y for m in measurements]
```

Comment:
This is the second time we need this (see other occurrence in `analyze_and_offset_measurements()`), so it might be a good idea to put this as method into the measurement itself, e.g. `x,y = m.get_data()` or something like that.